cg/meta/rsync.py

```diff
@@ -22,30 +25,51 @@ def __init__(self, config: CGConfig):
         self.destination_path: str = config.data_delivery.destination_path
         self.covid_destination_path: str = config.data_delivery.covid_destination_path
         self.covid_report_path: str = config.data_delivery.covid_report_path
-        self._process = None
-
-    @property
-    def process(self):
-        if not self._process:
-            self._process = Process("rsync")
-        return self._process
-
-    def run_rsync_command(self, ticket_id: int = None, dry_run: bool = False) -> None:
+        self.base_path: str = config.data_delivery.base_path
+        self.account: str = config.data_delivery.account
+        self.mail_user: str = config.data_delivery.mail_user
 
+    def get_all_cases_from_ticket(self, ticket_id: int) -> List[models.Family]:
         cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        return cases
+
+    def get_source_path(self, ticket_id: int) -> str:
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
         if not cases:
             LOG.warning("Could not find any cases for ticket_id %s", ticket_id)
             raise CgError()
-
         customer_id: str = cases[0].customer.internal_id
-        rsync_destination_path: str = self.destination_path % (customer_id, ticket_id)
         delivery_source_path: str = (
             Path(self.delivery_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/"
         )
-        rsync_options = "-rvL"
-        parameters: List[str] = [rsync_options, delivery_source_path, rsync_destination_path]
-        self.process.run_command(parameters=parameters, dry_run=dry_run)
+        return delivery_source_path
+
+    def get_destination_path(self, ticket_id: int) -> str:
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        customer_id: str = cases[0].customer.internal_id
+        rsync_destination_path: str = self.destination_path % (customer_id, ticket_id)
```

Comment:
I think it is more readable to create a function something like 'def get_rsync_destination_path(customer_id, ticket_id): 
  return Path(self.delivery_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/" 
because you as a programmer can read it and determine its validity. 

A "recept" that is read from a configuration can not be read and validated by a programmer as it is not part of the source. I suggest you move the "recept" in the code

Reply:
I have now updated this (and done necessary updates in the servers repo)